File: huipyo/src/week01/boj3986/Main.java
package week01.boj3986;

import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int answer = 0;

        for (int i = 0; i < n; i++) {
            String s = scanner.next();
            LinkedList<Character> stack = new LinkedList<>();

            for (char ch : s.toCharArray()) {
                if (!stack.isEmpty() && stack.peek() == ch) stack.pop();
                else stack.push(ch);
            }
            if (stack.isEmpty()) answer++;
        }
        System.out.println(answer);
        scanner.close();
    }
}

File: huipyo/src/week01/boj3986/Main.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on("n", (n) => {
    let answer = 0;

    for (let i = 0; i < parseInt(n); i++) {
        rl.question('', (s) => {
            var stack = [];

            for (let ch of s) {
                if (stack.length !== 0 && stack[stack.length - 1] === ch) stack.pop();
                else stack.push(ch);
            }
            if (stack.length === 0) answer++;
            console.log(answer);
            rl.close();
        }
        });
    }
});